integration-test-groups/foundation/language/src/test/java/org/apache/camel/quarkus/component/language/it/LanguageTest.java

```diff
@@ -16,35 +16,216 @@
+
+            /* Test transform=false option */
+            Map.entry("languageSimpleTransform", new TestResult("SimpleTransform", "SimpleTransform"))
+
+    );
+
+    private static Collection<Object[]> getLanguageResultsMap() {
+        return languageResultsMap.entrySet()
+                .stream()
+                .map((entry) -> new Object[] { entry.getKey(), entry.getValue() })
+                .collect(Collectors.toList());
+    }
+
+    private static final String OUTPUT_DIRECTORY = "target";
+    private static final String TEST_FILE = "test-file.txt";
+    private static final String SIMPLE_FILE = "hello.simple-file.txt";
+
+    @BeforeAll
+    public static void setupTestFiles() throws Exception {
+        Files.createDirectories(Paths.get(OUTPUT_DIRECTORY));
+        // Write the file used in the 'file' language test
+        Files.writeString(Paths.get(OUTPUT_DIRECTORY, TEST_FILE), "Dummy text", StandardCharsets.UTF_8);
+
+        // Copy the simple script from resources to the OUTPUT_DIRECTORY to test reading a script from a file: resource
+        Files.copy(
+                LanguageTest.class.getClassLoader().getResourceAsStream(SIMPLE_FILE),
+                Paths.get(OUTPUT_DIRECTORY, SIMPLE_FILE),
+                StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    @AfterAll
+    public static void deleteTestFiles() throws Exception {
+        Files.delete(Paths.get(OUTPUT_DIRECTORY, TEST_FILE));
+        Files.delete(Paths.get(OUTPUT_DIRECTORY, SIMPLE_FILE));
+    }
+
+    /**
+     * This test is called for each entry in {@link languageResultsMap}.
+     * The body text is passed to the routeName endpoint, which is checked for successful completion
+     * and that the returned body matches the expected result from {@link TestResult}
+     * 
+     * @param routeName - The name of the endpoint exposed in {@link LanguageResource}
+     * @param result    - The {@link TestResource} containing input body and expected result
+     */
+    @ParameterizedTest
+    @MethodSource("getLanguageResultsMap")
+    public void testLanguage(String routeName, TestResult result) {
         RestAssured.given()
                 .contentType(ContentType.TEXT)
-                .body("Dolly")
-                .post("/language/route/languageSimpleScript")
+                .body(result.body)
+                .post("/language/route/" + routeName)
                 .then()
                 .statusCode(200)
-                .body(Matchers.is("Hello Dolly from simple language script"));
+                .body(Matchers.is(result.expected));
     }
 
+    /**
+     * Tests whether scripts can be loaded over HTTP.
+     * 
+     * @throws Exception
+     */
     @Test
-    public void languageSimpleResource() {
+    public void testHttpResource() throws Exception {
+        // Stub the WireMock server to return a simple script on the "/simple" endpoint
+        assertNotNull(server);
+        server.stubFor(request("GET", urlPathEqualTo("/simple"))
```

Comment:
FYI, when using `WireMockTestResourceLifecycleManager`, we usually put Wiremock requests/responses in json files under resources/mappings, unless we have a reason why we do the stub manually (and we comment about it).
The stub are automatically created. 

Example: https://github.com/apache/camel-quarkus/tree/main/integration-tests/digitalocean/src/test/resources/mappings

Reply:
In this case, there is no existing REST service to mock. I just needed to test loading a resource over HTTP and figured a WireMock server would do the job. This is also handy in the contentCache test to ensure that the endpoints are called the expected number of times. However, I'm happy to add more comments to clarify this use case as it doesn't fit the pattern of other uses.